### PR TITLE
style: Minor cleanups in the rule tree code.

### PR DESCRIPTION
**Pull request by emilio**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16974)

### CODE REVIEW

File: components/style/rule_tree/mod.rs [hunk 1]
```diff
@@ -684,7 +684,7 @@ impl StrongRuleNode {
         let mut last = None;
         // TODO(emilio): We could avoid all the refcount churn here.
         for child in self.get().iter_children() {
-            if child .get().level == level &&
+            if child.get().level == level &&
```
Comment (stshine) [inline components/style/rule_tree/mod.rs]: good :)

Comment (emilio) [inline components/style/rule_tree/mod.rs]: Heh, it's been a long time :)

File: components/style/rule_tree/mod.rs [hunk 2]
```diff
@@ -855,20 +856,15 @@ impl StrongRuleNode {
...
-                let _ = Box::from_raw(weak.ptr());
+            let node = &*weak.ptr();
+            if node.refcount.load(Ordering::Relaxed) != 0 {
+                // Nothing to do, the node is still alive.
+                continue;
             }
+
+            debug!("GC'ing {:?}", weak.ptr());
+            node.remove_from_child_list();
+            let _ = Box::from_raw(weak.ptr());
```
Comment (nox) [inline components/style/rule_tree/mod.rs]: I don't like this rewrite. `node` is dangling after this line.

Comment (emilio) [inline components/style/rule_tree/mod.rs]: Sure, but the loop ends there, right? I can revert it with a comment if you prefer.

Comment (eddyb) [inline components/style/rule_tree/mod.rs]: `node` is a reference and not used past this point. @nikomatsakis or @arielb1 can confirm.

**Comment by nox** [comment]
@bors-servo retry

```
LLVM ERROR: Invalid data was encountered while parsing the file
```

**Pull request merged**